### PR TITLE
Refactored logs to custom logger

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: src/middlewares/error_handling.js [hunk 1]
```diff
@@ -14,18 +14,32 @@ import {
...
+    status = 400;
   } else if (err instanceof AuthenticationError) {
-    res.status(401).send({reason: err.message});
+    status = 401;
   } else if (err instanceof PermissionError) {
-    res.status(403).send({reason: err.message});
+    status = 403;
   } else if (err instanceof NotFoundError) {
-    res.status(404).send({reason: err.message});
+    status = 404;
```
Comment (spherefoundry) [inline src/middlewares/error_handling.js]: :+1:

File: src/start.js [hunk 2]
```diff
@@ -8,17 +8,20 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
...
+  const application = new Application(logger);
   await application.build();
   await application.ensureAdminAccountExist();
   await application.startServer();
   await application.startBackground();
 }
 
+const logger = new WinstonConsoleLogger();
+
 start()
```
Comment (spherefoundry) [inline src/start.js]: missing logger parameter for start method

File: test/middlewares/error_handling.js [hunk 3]
```diff
@@ -22,49 +22,57 @@ describe('Error handling middleware', () => {
...
+    };
     request = httpMocks.createRequest({});
     response = httpMocks.createResponse();
     next = sinon.spy();
   });
 
 
   it('should return 400 if ValidationError', async () => {
-    errorHandling(new ValidationError(), request, response, next);
+    errorHandling(mockLogger)(new ValidationError(), request, response, next);
```
Comment (spherefoundry) [inline test/middlewares/error_handling.js]: instead of instantiating a middleware each time, you could do it once in the beforeEach

**Comment by danihodovic** [comment]
Nice :clap: :clap: :clap:

**Pull request merged**